### PR TITLE
Read functionality

### PR DESCRIPTION
**Pull request by NTR-TheMekaVerse**

### CODE REVIEW

File: app/javascript/components/components/Header.js [hunk 1]
```diff
@@ -33,11 +36,13 @@ class Header extends Component {
...
                                 </NavLink>
                             </NavItem>
+
                             <NavItem>
-                                <NavLink href="/">
-                                    <img src={home_image} alt="home link" />
-                                </NavLink>
+                                    <NavLink href={homeLink}>
+                                        <img src={home_image} alt="home link" />
+                                    </NavLink>
```
Comment (zan-rosenthal) [inline app/javascript/components/components/Header.js:43]: This seems like it's indented too far. Can move it back 2 spaces

File: app/javascript/components/pages/TaskIndex.js [hunk 2]
```diff
@@ -1,10 +1,38 @@
...
+                    <h1>Task Index</h1>
+                    <br />
+                    <a href="/task_new" >
+                        <Button color="primary">
+                            Create New Task
+                        </Button>
+                    </a>
+                    <br />
+
+                    {this.props.tasks && this.props.tasks.map(task => {
```
Comment (zan-rosenthal) [inline app/javascript/components/pages/TaskIndex.js:20]: You can use optional chaining operator to handle the null checking here, like this: `this.props.tasks?.map`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

**Review by zan-rosenthal — CHANGES_REQUESTED** [review]

File: app/javascript/components/components/Header.js [hunk 3]
```diff
@@ -21,7 +24,7 @@ class Header extends Component {
                         expand="md"
                         light
                     >
-                        <NavbarBrand href="/">
+                        <NavbarBrand href={homeLink}>
```
Comment (sjproctor) [inline app/javascript/components/components/Header.js:27]: You may run into issues using an `href` for internal React navigation. It is recommended to use `NavLink to=` from react-router-dom

File: app/javascript/components/pages/TaskIndex.js [hunk 4]
```diff
@@ -1,10 +1,38 @@
...
 class TaskIndex extends Component {
+
     render() {
         return (
             <>
-                <h1>TaskIndex</h1>
+                <Container>
+                    <h1>Task Index</h1>
+                    <br />
+                    <a href="/task_new" >
```
Comment (sjproctor) [inline app/javascript/components/pages/TaskIndex.js:13]: This should also be a NavLink

File: app/javascript/components/components/Header.js [hunk 5]
```diff
@@ -33,11 +36,13 @@ class Header extends Component {
                                     ABOUT US
                                 </NavLink>
                             </NavItem>
+
                             <NavItem>
-                                <NavLink href="/">
-                                    <img src={home_image} alt="home link" />
-                                </NavLink>
+                                    <NavLink href={homeLink}>
```
Comment (sjproctor) [inline app/javascript/components/components/Header.js:41]: Same thing here.

**Review by sjproctor — CHANGES_REQUESTED** [review]

**Review by zan-rosenthal — APPROVED** [review]

**Review by sjproctor — APPROVED** [review]

**Pull request merged**